lib/Optimizer/CodeGen/QuakeToLLVM.cpp

```diff
@@ -1213,6 +1214,137 @@ class RemoveRelaxSizeRewrite : public OpConversionPattern<quake::RelaxSizeOp> {
     return success();
   }
 };
+
+class CustomUnitaryOpRewrite
+    : public ConvertOpToLLVMPattern<quake::CustomUnitarySymbolOp> {
+public:
+  using ConvertOpToLLVMPattern::ConvertOpToLLVMPattern;
+
+  /// Function to convert a QIR Qubit value to an Array value.
+  /// TODO: Refactor to reuse code from 'ConcatOpRewrite' (line#247)
+  Value wrapQubitInArray(Location &loc, ConversionPatternRewriter &rewriter,
+                         ModuleOp parentModule, Value v) const {
+    auto context = rewriter.getContext();
+    auto qirArrayTy = cudaq::opt::getArrayType(context);
+    auto i8PtrTy = cudaq::opt::factory::getPointerType(context);
+    FlatSymbolRefAttr symbolRef = cudaq::opt::factory::createLLVMFunctionSymbol(
+        cudaq::opt::QIRArrayCreateArray, qirArrayTy,
+        {rewriter.getI32Type(), rewriter.getI64Type()}, parentModule);
+    FlatSymbolRefAttr getSymbolRef =
+        cudaq::opt::factory::createLLVMFunctionSymbol(
+            cudaq::opt::QIRArrayGetElementPtr1d, i8PtrTy,
+            {qirArrayTy, rewriter.getIntegerType(64)}, parentModule);
+    Value zero = rewriter.create<arith::ConstantIntOp>(loc, 0, 64);
+    Value one = rewriter.create<arith::ConstantIntOp>(loc, 1, 64);
+    // FIXME: 8 bytes is assumed to be the sizeof(char*) on the target machine.
+    Value eight = rewriter.create<arith::ConstantIntOp>(loc, 8, 32);
```

Comment:
Can we make both arguments 64 bit in the call below?

Reply:
Addressed in 9ded69b1946782c6a7e42876173de7eabf75ca68